src/CreateEntityMappingView/CreateEntityMappingView.js

```diff
@@ -249,98 +296,189 @@ class CreateEntityMappingView extends ConnectionComponent {
+	}
+
+	renderCreateView() {
+		const optionalActions = this.props.createEntityMappingResponse;
+		CreateEntityMappingView.handleOptionalActionsSuccess(optionalActions);
+		const initialDataLoaded = this.fetchInitialData();
+		if (!initialDataLoaded.hasOwnProperty("eventTypes")) {
+			return initialDataLoaded;
 		}
 		return (
 			<div>
 				<Header title={"Create Entity Mapping"}/>
 				<div className={AppStyles.elementMarginTop}>
-					<Container>
-						{optionalActions && optionalActions.rejected &&
-							<ErrorMessage message={optionalActions.reason} />
-						}
-						<Row>
-							<Col md={12}>
-								<SelectField
-									value={this.state.targetStatus}
-									onChange={this.handleTargetStatusChange}
-									floatingLabelText={"Target Status, leave empty if no status update is required"}
-									fullWidth={true}>
-									{config.statuses.map((status, key) => {
-										if (status.name !== "UNDEFINED") {
-                                            return <MenuItem
-												key={key}
-												value={status.name}
-												primaryText={status.name}/>;
-                                        }
-                                        return <div key={key}/>;
-									})}
-								</SelectField>
-							</Col>
-						</Row>
-						<Row>
-							<Col md={6}>
-								<SelectField
-									value={this.state.selectedEventTypeId.value}
-									errorText={this.state.selectedEventTypeId.errorMessage}
-									onChange={this.handleEventTypeChange}
-									floatingLabelText="Select Event Type"
-									fullWidth={true}>
-									{CreateEntityMappingView.getMenuItems(eventTypes)}
-								</SelectField>
-							</Col>
-							<Col md={6}>
-								<SelectField
-									value={this.state.selectedEntityTypeId.value}
-									errorText={this.state.selectedEntityTypeId.errorMessage}
-									onChange={this.handleEntityTypeChange}
-									floatingLabelText="Select Entity Type"
-									fullWidth={true}>
-									{CreateEntityMappingView.getMenuItems(entityTypes)}
-								</SelectField>
-							</Col>
-						</Row>
-						{attributeFields}
-						<div className={css(AppStyles.textAlignCenter)}>
-							<RaisedButton
-								label="Abort"
-								icon={<IconCancel/>}
-								className={css(AppStyles.marginAllSites)}
-								secondary={true}
-								onTouchTap={this.abort}
-							/>
-							<RaisedButton
-								label="Save"
-								icon={<IconSave/>}
-								onTouchTap={this.submitMapping}
-								className={css(AppStyles.marginAllSites)}
-								primary={true}
-							/>
-						</div>
-					</Container>
+					{this.renderMappingForm(initialDataLoaded, optionalActions, this.submitNewMapping)}
 				</div>
 			</div>
 		);
 	}
+
+	// loads entity mapping in edit view
+	componentWillMount() {
+		if (!this.isCreateView) {
+			this.props.lazyLoadEntityMapping();
+		}
+	}
+
+	componentDidUpdate(prevProps, prevState) {
+		if (!this.isCreateView && !this.oldValuesWereLoaded && this.oldValuesShouldBeLoaded) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Merge this if statement with the nested one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3ACollapsibleIfStatements)